fe/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java

```diff
@@ -0,0 +1,259 @@
+                iterator.remove();
+                continue;
+            }
+            String tableName;
+            ArrayList<AddPartitionClause> addPartitionClauses = new ArrayList<>();
+            db.readLock();
+            try {
+                // Only OlapTable has DynamicPartitionProperty
+                OlapTable olapTable = (OlapTable) db.getTable(tableId);
+                if (olapTable == null
+                        || !olapTable.dynamicPartitionExists()
+                        || !olapTable.getTableProperty().getDynamicPartitionProperty().getEnable()) {
+                    iterator.remove();
+                    continue;
+                }
+
+                // Determine the partition column type
+                // if column type is Date, format partition name as yyyyMMdd
+                // if column type is DateTime, format partition name as yyyyMMddHHssmm
+                // scheduler time should be record even no partition added
+                createOrUpdateRuntimeInfo(olapTable.getName(), LAST_SCHEDULER_TIME, TimeUtils.getCurrentFormatTime());
+                RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) olapTable.getPartitionInfo();
+                Column partitionColumn = rangePartitionInfo.getPartitionColumns().get(0);
+                String partitionFormat;
+                try {
+                    partitionFormat = DynamicPartitionUtil.getPartitionFormat(partitionColumn);
+                } catch (DdlException e) {
+                    recordFailedMsg(olapTable.getName(), e.getMessage());
+                    continue;
+                }
+
+                Calendar calendar = Calendar.getInstance();
+                TableProperty tableProperty = olapTable.getTableProperty();
+                DynamicPartitionProperty dynamicPartitionProperty = tableProperty.getDynamicPartitionProperty();
+
+                for (int i = 0; i <= dynamicPartitionProperty.getEnd(); i++) {
+                    String dynamicPartitionPrefix = dynamicPartitionProperty.getPrefix();
+                    String partitionRange = DynamicPartitionUtil.getPartitionRange(dynamicPartitionProperty.getTimeUnit(),
+                            i, (Calendar) calendar.clone(), partitionFormat);
+                    String partitionName = dynamicPartitionPrefix + DynamicPartitionUtil.getFormattedPartitionName(partitionRange);
+
+                    // continue if partition already exists
+                    String nextBorder = DynamicPartitionUtil.getPartitionRange(dynamicPartitionProperty.getTimeUnit(),
+                            i + 1, (Calendar) calendar.clone(), partitionFormat);
+                    PartitionInfo partitionInfo = olapTable.getPartitionInfo();
+                    boolean isPartitionExists = false;
+
+                    RangePartitionInfo info = (RangePartitionInfo) (partitionInfo);
+                    for (Range<PartitionKey> partitionKeyRange : info.getIdToRange().values()) {
```

Comment:
I think you should create a real `Range` object and check if it is intersect with existing ranges. That is more safe and reasonable.